tests/test_kernels.py

```diff
@@ -104,111 +122,42 @@ def test_required_grid_vars(grid_type_field_and_extra_kwargs):
-        laplacian = LaplacianClass(**test_kwargs)
-        diffused = laplacian(delta)
-        # check that delta function gets diffused isotropically in y-direction
-        np.testing.assert_allclose(
-            diffused[random_yloc - 1, random_xloc],
-            diffused[random_yloc + 1, random_xloc],
-            atol=1e-12,
-        )
+    if grid_type not in irregular_grids:
+        pytest.skip("This test is only for irregular grids")
 
+    # deploy mass at random location away from Antarctica
+    delta = np.zeros_like(data)
+    ny, nx = delta.shape
+    # pick a location outside of the mask
+    random_yloc = 99
+    random_xloc = 225
+    delta[random_yloc, random_xloc] = 1
 
-def test_flux_in_x_direction(grid_type_field_and_extra_kwargs):
-    """This test checks that the Laplacian computes the correct fluxes in x-direction if the grid is irregular.
-    The test will catch sign errors in the Laplacian rolling of array elements along the x-axis."""
-    grid_type, data, extra_kwargs = grid_type_field_and_extra_kwargs
+    LaplacianClass = ALL_KERNELS[grid_type]
+    laplacian = LaplacianClass(**extra_kwargs)
+    diffused = laplacian(delta)
 
-    if grid_type in irregular_grids:
-        # deploy mass at random location away from Antarctica: delta_{j,i}
-        delta = np.zeros_like(data)
-        ny = np.shape(delta)[0]
-        random_yloc = np.random.randint(5, ny)
-        nx = np.shape(delta)[1]
-        random_xloc = np.random.randint(2, nx - 2)
-        delta[random_yloc, random_xloc] = 1
-
-        test_kwargs = copy.deepcopy(extra_kwargs)
-        ones = np.ones_like(data)
-        # create highly irregular grid data for
-        # - western edge of cell {j,i-1};
-        # - western edge of cell {j,i+2}
-        # this should leave laplacian(delta_{j,i}) unaffected
-        if grid_type == GridType.IRREGULAR_WITH_LAND:
-            test_kwargs["wet_mask"] = ones  # no land for simplicity
-            test_kwargs["area"] = ones
-            test_kwargs["dys"] = ones
-            test_kwargs["dxw"] = ones
-            test_kwargs["dxs"] = ones
-            test_kwargs["dyw"] = ones.copy()
-            test_kwargs["dyw"][:, random_xloc - 1] = 1000
-            test_kwargs["dyw"][:, random_xloc + 2] = 2000
-        if grid_type == GridType.TRIPOLAR_POP_WITH_LAND:
-            test_kwargs["wet_mask"] = ones.copy()
-            test_kwargs["wet_mask"][0, :] = 0  # Antarctica
-            test_kwargs["tarea"] = ones
-            test_kwargs["dyn"] = ones
-            test_kwargs["dxe"] = ones
-            test_kwargs["dxn"] = ones
-            test_kwargs["dye"] = ones.copy()
-            test_kwargs["dye"][:, random_xloc - 2] = 1000
-            test_kwargs["dye"][:, random_xloc + 1] = 2000
+    # check that delta function gets diffused isotropically in y-direction
+    np.testing.assert_allclose(
+        diffused[random_yloc - 1, random_xloc],
+        diffused[random_yloc + 1, random_xloc],
```

Comment:
I was thinking this should be true in general for any irregular grid. However, the tests are currently failing, e.g.

```
test_flux[GridType.IRREGULAR_WITH_LAND]
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=1e-12
E       
E       Mismatched elements: 1 / 1 (100%)
E       Max absolute difference: 1.55178064
E       Max relative difference: 0.62167217
E        x: array(0.944359)
E        y: array(2.49614)
```

It doesn't make sense to me that the concentration could _ever_ be greater than 1, but that's what we see in the second value.

Any insights into what might be going wrong here?

Reply:
Small update, in 51121b3 I modified this test to use a constant area=1 and also reduced the amplitude of the randomness. But the test is still failing, albeit with different values

```
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=1e-12
E       
E       Mismatched elements: 1 / 1 (100%)
E       Max absolute difference: 0.18457325
E       Max relative difference: 0.20314357
E        x: array(1.093159)
E        y: array(0.908585)
```

I definitely feel stuck on this.

Reply:
I think values can be greater than 1 if your are on a irregular grid. Here is an example of how a flux across a northern cell edge computed:
```
 nflux = ((np.roll(data, -1, axis=-2) - data) / self.dyn * self.dxn)
```
If `self.dyn < 1` and `self.dxn >1`, then `nflux` will be larger than 1 if you consider the northern edge of the grid cell that the delta function lives in. The grid cell to the north does not get any flux contributions from other neighboring cells, so the concentration in this cell will be greater than 1 (for `area = 1`). Definitely wouldn't happen on a regular grid.